### PR TITLE
[networks] Improve system-probe kitchen test output

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Improves the output of system-probe kitchen tests.
Failures are now consolidated in the test summary like the following:

```
FAILURES:

system-probe tests for /tmp/system-probe-tests/pkg/ebpf/bytecode/testsuite succesfully runs:
--- FAIL: TestEbpfBytesCorrect (0.02s)

system-probe tests for /tmp/system-probe-tests/pkg/network/testsuite succesfully runs:
--- FAIL: TestDNSOverUDPTimeoutCount (5.16s)

Finished in 1 minute 16.19 seconds.
Platform: Linux dd-system-probe-test-xenial 4.4.0-193-generic #224-Ubuntu SMP Tue Oct 6 17:15:28 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

In addition to that we're no longer relying on the exit code of the `go test` tool which has been unreliable in certain cases.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/ebpf/bytecode/ebpf_bytes_match_test.go [hunk 1]
```diff
@@ -25,6 +26,7 @@ func TestEbpfBytesCorrect(t *testing.T) {
 		actual, err := ioutil.ReadAll(actualReader)
 		require.NoError(t, err)
 
-		assert.Equal(t, bs, actual, fmt.Sprintf("on-disk file %s and bundled content are different", filename))
```
Comment (p-lambert) [inline pkg/ebpf/bytecode/ebpf_bytes_match_test.go:28]: This was generating a huge hex dump during failures

File: test/kitchen/test/integration/dd-system-probe-test/rspec/dd-system-probe-test_spec.rb [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
-        retval = $?
-        expect(retval).to eq(0)
+        Open3.popen2e("sudo", f, "-test.v") do |_, output, _|
+          test_failures = []
+          output.each_line do |line|
+            puts "#{line}"
+            test_failures << line.strip if line =~ GOLANG_TEST_FAILURE
+          end
+
+          expect(test_failures).to be_empty, test_failures.join("\n")
```
Comment (brycekahle) [inline test/kitchen/test/integration/dd-system-probe-test/rspec/dd-system-probe-test_spec.rb:20]: Can we keep the return value check as well? I imagine there are scenarios where some the test binary fails immediately that wouldn't have `FAIL:` in the output?

Comment (p-lambert) [reply test/kitchen/test/integration/dd-system-probe-test/rspec/dd-system-probe-test_spec.rb:20]: Can do

**Review by brycekahle — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]
LGTM.

I think `/test/kitchen/test/integration/dd-system-probe-test/` should be owned by networks in the `CODEOWNERS` file, since it only contains test code specific to the system-probe kitchen tests (and `/test/kitchen/test/integration/dd-system-probe-test/dd-security-agent-test/` should be owned by agent-security for similar reasons). WDYT?

**Pull request merged**